homeassistant/components/ukraine_alarm/__init__.py

```diff
@@ -0,0 +1,107 @@
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    ALERT_TYPE_AIR,
+    ALERT_TYPE_ARTILLERY,
+    ALERT_TYPE_UNKNOWN,
+    ALERT_TYPE_URBAN_FIGHTS,
+    DOMAIN,
+    PLATFORMS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+UPDATE_INTERVAL = timedelta(seconds=10)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Ukraine Alarm as config entry."""
+    api_key = entry.data[CONF_API_KEY]
+    region_id = entry.data.get(CONF_REGION)
+
+    websession = async_get_clientsession(hass)
+
+    coordinator = UkraineAlarmDataUpdateCoordinator(
+        hass, websession, api_key, region_id
+    )
+    await coordinator.async_config_entry_first_refresh()
+
+    entry.async_on_unload(entry.add_update_listener(update_listener))
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Update options."""
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class UkraineAlarmDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Class to manage fetching Ukraine Alarm API."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        session: ClientSession,
+        api_key,
+        region_id,
+    ) -> None:
+        """Initialize."""
+        self.region_id = region_id
+        self.ukrainealarm = Client(session, api_key)
+
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=UPDATE_INTERVAL)
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        """Update data via library."""
+        try:
+            res = await self.ukrainealarm.get_alerts(self.region_id)
+        except aiohttp.ClientError as error:
+            raise UpdateFailed(f"Error fetching alerts from API: {error}") from error
+
+        current = {
+            ALERT_TYPE_AIR: False,
+            ALERT_TYPE_UNKNOWN: False,
+            ALERT_TYPE_ARTILLERY: False,
+            ALERT_TYPE_URBAN_FIGHTS: False,
+        }
+        for alert in res[0]["activeAlerts"]:
+            if alert["type"] == ALERT_TYPE_AIR:
+                current[ALERT_TYPE_AIR] = True
+            if alert["type"] == ALERT_TYPE_UNKNOWN:
+                current[ALERT_TYPE_UNKNOWN] = True
+            if alert["type"] == ALERT_TYPE_ARTILLERY:
+                current[ALERT_TYPE_ARTILLERY] = True
+            if alert["type"] == ALERT_TYPE_URBAN_FIGHTS:
+                current[ALERT_TYPE_URBAN_FIGHTS] = True
```

Comment:
You can simplify this logic a bit with
https://github.com/bdraco/home-assistant/commit/f1b50d11abfddf75461c03536db29cf8b3180018

Reply:
Agree. Updated.